File: FizzBuzz-Java/src/jp/sf/fizzbuzz/FizzBuzz.java
/**
 *
 */
package jp.sf.fizzbuzz;

/**
 * @author sf
 * FizzBuzzの結果を計算するクラス
 */
public class FizzBuzz {

    /**
     * @author sf
     * FizzBuzzと宣言する値と宣言時の言葉を組みで管理するクラス
     * とりあえず一般的なやつを定数として保持する
     */
    public enum FIZBUZWORD {

        FIZZ("FIZZ",3),
        BUZZ("BUZZ",5);

        private final String word;
        private final Integer number;

        FIZBUZWORD(final String w, final Integer n){
            word = w;
            number = n;
        }

        public Integer getNumber(){
            return number;
        }

        public String getWord(){
            return word;
        }
    }

    public String verifyFizzBuzz(Integer value){

        String result = ""; // 初期文字列は空文字とする
        // 特定ワードについて数字で割ってみる
        for(FIZBUZWORD oneNumber : FIZBUZWORD.values()){
            // 割り切れたら文字を追加していく
            if(0 == (value % oneNumber.getNumber())){
                result += oneNumber.getWord();
            }
        }
        // 一度も文字が追加されていない場合は、数字で返答する
        if("".equals(result)) result = value.toString();

        return result;
    }
}

File: FizzBuzz-JS/FizzBuzz.js

/**
 * @param word 宣言する言葉
 * @param number 宣言時の数字
 * FizzBuzz判定時の宣言する言葉と判定時の数字
 */
function FIZZBUZZWORD(word, number){
  this.word = word;
  this.number = number;
  return (this);
}

var FBWORDLIST = new Array(2);
FBWORDLIST[0] = new FIZZBUZZWORD("FIZZ",3);
FBWORDLIST[1] = new FIZZBUZZWORD("BUZZ",5);

/**
 * @param value FIZZBUZZ判定する値
 * FizzBuzz判定をする
 */
function verifyFizzBuzz(value){

  // 初期文字列は空文字にしておく
  var result = "";
  // 特定ワードについて数字で割ってみる
  for (var i = 0; i < FBWORDLIST.length ; i++ ){
    var w = FBWORDLIST[i];
    // 割り切れたら文字を追加する
    if(0 == (value % w.number)){
      result += w.word;
    }
  }
  // 一度も文字が追加されていない場合は数字で返答する
  if("" == result) result = value.toString();

  return (result);

}